eng/common/scripts/get-codeowners.ps1

```diff
@@ -14,30 +17,105 @@ if (!(Test-Path $codeOwnersLocation)) {
     $splitLine = $contentLine -split "\s+"
     
     # CODEOWNERS file can also have labels present after the owner aliases
     # gh aliases start with @ in codeowners. don't pass on to API calls
-    $ownedFolders[$splitLine[0].ToLower().Trim("/")] = ($splitLine[1..$($splitLine.Length)] `
-      | ? { $_.StartsWith("@") } `
-      | % { return $_.substring(1) }) -join ","
+
+    $aliases = ($splitLine[1..$($splitLine.Length)] | ? { $_.StartsWith("@") } | % { return $_.substring(1) }) -join ","
+    $labels = ""
+
+    if ($null -ne $previousLine -and $previousLine.Contains("PRLabel:"))
+    {
+      $previousLine = $previousLine.substring($previousLine.IndexOf(':') + 1)
+      $splitPrevLine = $previousLine -split "%" 
+      $labels = ($splitPrevLine[1..$($splitPrevLine.Length)] | % { return $_.Trim() }) -join ","
+    }
+
+    $ownedFolders[$splitLine[0].ToLower().Trim("/")] = @{ Aliases = $aliases; Labels = $labels }
   }
+  $previousLine = $contentLine
 }
 
 $results = $ownedFolders[$target]
 
 if ($results) {
-  Write-Host "Found a folder $results to match $target"
-  
-  if ($VsoVariable) {
-    $alreadyPresent = [System.Environment]::GetEnvironmentVariable($VsoVariable)
+  Write-Host "Found a folder to match $target"
+  $aliases = $results.Aliases -split ","
+  $users = @()
+  $teams = @()
+
+  foreach ($str in $aliases)
+  {
+    if ($str.IndexOf('/') -ne -1) # Check if it's a team alias e.g. Azure/azure-sdk-eng
+    {
+      $org = $str.substring(0, $str.IndexOf('/'))
+      $team_slug = $str.substring($str.IndexOf('/') + 1)
+      $teamApiUrl =  "https://api.github.com/orgs/$org/teams/$team_slug"
+      if (VerifyAlias -APIUrl $teamApiUrl)
+      {
+        $teams += $team_slug
+        continue
+      }
+    }
+    else
+    {
+      $usersApiUrl = "https://api.github.com/users/$str"
+      if (VerifyAlias -APIUrl $usersApiUrl)
+      {
+        $users += $str
+        continue
+      }
+      Write-Host "Alias ${str} is neither a recognized github user nor a team"
+    }
+  }
+
+  $labelsString = $results.Labels
+
+  if ($VsoOwningUsers) {
+    $presentOwningUsers = [System.Environment]::GetEnvironmentVariable($VsoOwningUsers)
+    if ($presentOwningUsers) { 
+      $user += $presentOwningUsers
+    }
+    Write-Host "##vso[task.setvariable variable=$VsoOwningUsers;]($user -join ',')"
+  }
+
+  if ($VsoOwningTeams) {
+    $presentOwningTeams = [System.Environment]::GetEnvironmentVariable($VsoOwningTeams)
+    if ($presentOwningTeams) { 
+      $teams += $presentOwningUsers
+    }
+    Write-Host "##vso[task.setvariable variable=$VsoOwningTeams;]($teams -join ',')"
+  }
 
-    if ($alreadyPresent) { 
-      $results += ",$alreadyPresent"
+  if ($VsoOwningLabels) {
+    $presentOwningLabels = [System.Environment]::GetEnvironmentVariable($VsoOwningLabels)
+    if ($presentOwningLabels) {
+      if ($labelsString) { $labelsString += ",$presentOwningLabels" } else { $teamsString = "$presentOwningLabels" }
```

Comment:
Once $labels is an array just do a += like the users and teams.

Reply:
Does not seem like this will work if the returned value from the environment variable is already a comma separated string which I think is what @scbedd was anticipating.

Reply:
Never mind it'll all be fine when join is called.